spec-0006/index.md

```diff
@@ -0,0 +1,113 @@
+Most online services (GitHub, PyPi, etc.), have permissioning systems through which developers can be given the required access.
+Access to servers can be given through SSH keys, which each developer has to safeguard.
+This document deals with the situation in which that is not possible, and secrets have to be shared.
+
+## Implementation
+
+### Principles
+
+1. Restricted project resources must be documented.
+
+2. A system for distributing project secrets must have the following properties:
+
+   - Secrets are stored encrypted in a central (remote) location.
+   - It must be possible to grant access to the secrets to a select group of team members.
+   - It must be possible to revoke future access to the secrets.
+   - The system must not rely on closed source or commercial encryption facilities, that
+     can later disappear or be made unavailable, although such a solution can be considered if it allows for all data to be exported.
+
+   Whichever system is chosen, its user interface should match the capabilities of the intended users.
+   This reduces the risk of passwords being copied out to less secure mechanisms such as sticky notes or text files.
+   If the target audience is not used to GPG or the command prompt, for example, the `pass` implementation below may not work, and an alternative like 1password or bitwarden should be considered.
+
+3. _Tokens_ form a common subset of secrets stored. These should always be scoped (minimal necessary permissions), and set to expire after a reasonable time period (a year, typically).
+   Instructions for rotating and revoking the token should be documented.
+
+### Password storage
+
+#### Hosted
+
+The following hosted solutions conform to the principles in (2) above.
+
+Sponsored solutions:
+
+- [1password](https://github.com/1Password/1password-teams-open-source)
+
+Self-hosted solutions:
+
+- [vaultwarden](https://github.com/dani-garcia/vaultwarden)
+
+Paid solutions:
+
+- [bitwarden](https://bitwarden.com/) provides a non-profit discount of 25%
+
+#### Offline (pass)
+
+[password vault](https://github.com/scientific-python/vault-template) is an example of an implementation that satisfies the above principles.
+The secrets are stored, encrypted, in a public Git repository.
+The vault uses [gopass](https://github.com/gopasspw/gopass), a more user friendly implementation of [pass](https://www.passwordstore.org/), to manage access via GPG keys.
+Each secret is encrypted using the public keys of all developers that should have access.
+If a developer's access is removed, the vault is re-encrypted so that that developer cannot read future copies of the repository (but secrets should be considered compromised and, thus, rotated).
+
+### Other common scenarios
+
+- **Publishing packages**: PyPi provides a [trusted publisher](https://docs.pypi.org/trusted-publishers/using-a-publisher/) mechanism for avoiding passwords
+
+### Other security considerations
```

Comment:
This could also list a bullet point on what makes a secure secret in case of conventional passwords (e.g passphrases, minimal password length, etc). I imagine using the recommended password managers here, makes it easy to use long ones that don't have to be remembered and can just be copy-pasted. Or maybe just point out this fact.

Might be to detailed for the scope of this document though...